mitiq/rem/rem.py

```diff
@@ -0,0 +1,173 @@
+# Copyright (C) 2021 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Readout Confusion Inversion."""
+
+from typing import Callable, Iterable, List, Optional, Sequence, Tuple, Union
+
+from functools import partial, wraps
+import numpy as np
+
+from cirq.sim import sample_state_vector as sample_prob_dist
+from cirq.qis.states import to_valid_state_vector as to_prob_dist
```

Comment:
A state vector and a probability distribution are different mathematical objects. Roughly speaking there is a square root difference between the elements of a state and the element of a probability distribution. 

Isn't this a problem here? It seems that we are using functions that are defined for sampling and generating quantum states for a different task (sampling and generating probability distributions).

Reply:
I've replaced these calls with custom helper methods. You are correct about the semantics of this though. The calls from were Cirq were symmetric, so the result was still the same, but it's unnecessary to turn the measurement result back into a state vector.